pilot/pkg/model/context.go

```diff
@@ -318,9 +361,36 @@ func ParseServiceNodeWithMetadata(s string, metadata map[string]string) (*Proxy,
 
 	out.ID = parts[2]
 	out.DNSDomain = parts[3]
+	out.IstioVersion = ParseIstioVersion(metadata[NodeMetadataIstioVersion])
 	return out, nil
 }
 
+// ParseIstioVersion parses a version string and returns IstioVersion struct
+func ParseIstioVersion(ver string) *IstioVersion {
+	if strings.HasPrefix(ver, "master-") {
```

Comment:
I think in the future we should kill the `master-` and replace with proper semver. wdyt?

Reply:
I would love to.. same goes for release as well.